v2/internal/pkg/release/local_stored_collector.go

```diff
@@ -233,26 +233,22 @@ func (o LocalStorageCollector) prepareD2MCopyBatch(log clog.PluggableLoggerInter
 			o.Log.Error("%s", err.Error())
 			return nil, err
 		}
+		src = imgSpec.ReferenceWithTransport
 		if imgSpec.IsImageByDigest() {
-			src = dockerProtocol + strings.Join([]string{o.LocalStorageFQDN, imgSpec.PathComponent + "@" + imgSpec.Algorithm + ":" + imgSpec.Digest}, "/")
-			dest = strings.Join([]string{o.Opts.Destination, imgSpec.PathComponent + "@" + imgSpec.Algorithm + ":" + imgSpec.Digest}, "/")
+			tag := fmt.Sprintf("%s-%s", imgSpec.Algorithm, imgSpec.Digest)
+			dest = dockerProtocol + strings.Join([]string{o.destinationRegistry(), imgSpec.PathComponent + ":" + tag[:127]}, "/")
```

Comment:
@aguidirh maybe add a check to see if it is greater than 128 and then do the trim

Reply:
Done